modules/assim.sequential/inst/NEFI/generate.gefs.xml.R

```diff
@@ -0,0 +1,135 @@
+  forecast_hour = sprintf("%04d", forecast_hour * 100)
+  date = as.POSIXct(paste0(lubridate::year(date), "-", lubridate::month(date), "-", lubridate::day(date), " ", 
+                                 substring(forecast_hour, 1,2), ":00:00"), tz="UTC")
+  
+  return(date)
+}
+
+
+##'
+##'This script sets up the xml for, and runs PEcAn for the following settings:
+##'
+##'NOAA_GEFS data
+##'Start date: current date/time
+##'End date: 16 days in the future
+##'model: SIPNET
+##'
+##'@author Luke Dramko
+
+# Settings file (including path) is passed in on the command line.
+args <- commandArgs(trailingOnly = TRUE)
+if (length(args) < 1) {
+  print("Please include an xml file as a command line argument.")
+  quit("no", status = 10)
+}
+if (!file.exists(args[1])) {
+  print(paste0("File ", args[1], " does not exist."))
+  quit("no", status = 11)
+}
+if (!is.na(args[2])) {
+  source_date <- args[2]
+}
+
+# These lines from the PEcAn workflow load the settings object.
+settings <- PEcAn.settings::read.settings(args[1])
+
+xmloutdir <- regmatches(args[1], regexpr("(~|\\./)?(/)?([^/]*/)*", args[1])) # extract xmloutdir from args[1]
+filename <- basename(args[1]) # Extract file name from args[1]
+
+if(xmloutdir == "") {xmloutdir <- "."}
+
+dbparms = list()
+dbparms$dbname = "bety"
+dbparms$host = "128.197.168.114"
+dbparms$user = "bety"
+dbparms$password = "bety"
+
+#Connection code copied and pasted from met.process
+bety <- dplyr::src_postgres(dbname   = dbparms$dbname, 
+                            host     = dbparms$host, 
+                            user     = dbparms$user, 
+                            password = dbparms$password)
+
+con <- bety$con #Connection to the database.  dplyr returns a list.
+if (is.null(con)) {
+  print("Database connection failed.")
+  quit("no", status=12)
+}
+
+# Set the run dates
+if (exists("source_date")) {
+  start_date <- round.to.six.hours(source_date)
+} else {
+  start_date <- round.to.six.hours()
+}
+
+end_date <- start_date + lubridate::days(16)
+settings$run$start.date <- as.character(start_date)
+settings$run$end.date <- as.character(end_date)
+
+#settings$ensemble$start.year <- start_date
+#settings$ensemble$end.year <- end_date
+
+# Update the time of the run
+settings$info$date <- paste0(format(Sys.time(), "%Y/%m/%d %H:%M:%S"), " +0000")
+
+# Update met.start and met.end for the xml (shouldn't make a difference,
+# but good practice anyway).
+settings$run$site$met.start <- format(Sys.time() - lubridate::days(12), "%Y-%m-%d")
+settings$run$site$met.end <- format(Sys.time() + lubridate::days(16), "%Y-%m-%d")
+
+settings$ensemble$start.year <- format(start_date, "%Y")
+settings$ensemble$end.year <- as.character(end_date, "%Y")
+
+# Determine the workflow ID by looking in the database
+ids <- PEcAn.DB::db.query("SELECT id FROM workflows", con)
+ids <- ids[ids<2000000000] # There are several placeholder ids that are much too large in the database, so this screens
```

Comment:
FYI, it's more efficient to have SQL do this sort of filtering than to query a large table and then post-process it in R

Reply:
Also, IDs over 2e9 aren't placeholders, they're from [other nodes in the BETY network](https://pecan.gitbook.io/betydb-documentation/betydb-system-administration/distributed-betydb#primary-key-allocations). As written, this won't work for any workflow created outside the Dietze or Serbin labs.